be/src/util/memcmp.h

```diff
@@ -56,16 +56,17 @@ inline int compare(T lhs, T rhs) {
     }
 }
 
-// mem_equal is used to optimize the comparastion between the two strings.
+// mem_equal_safe is used to optimize the comparison between the two strings.
 //  1. If the length is equal and larger than 16, use SSE4.1
 //  2. If the length is small than 16, convert the address to int16/int32/int64
-//     to comparasion
+//     to comparison
+// so it is safe that strings do not need to consider extra padding bytes for SIMD.
 // TODO: If know the size in advance, call the function by constant parameter
 //       like memequal(p1, 10, p2, 10) is efficient
 
 #if defined(__SSE4_1__) && !defined(ADDRESS_SANITIZER)
 
-inline bool memequal(const char* p1, size_t size1, const char* p2, size_t size2) {
+inline bool memequal_safe(const char* p1, size_t size1, const char* p2, size_t size2) {
```

Comment:
Theoretically, This operation should be safe by itself, we don’t need to emphasize this point, we have an unsafe suffix  method is enough

Reply:
yeah, but a `_safe` reminder some new one that ther is a `_unsafe` memequal()

Reply:
renamed